libpod/container_internal.go

```diff
@@ -770,21 +835,29 @@ func (c *Container) makeBindMounts() error {
 // writeStringToRundir copies the provided file to the runtimedir
 func (c *Container) writeStringToRundir(destFile, output string) (string, error) {
 	destFileName := filepath.Join(c.state.RunDir, destFile)
+
+	if err := os.Remove(destFileName); err != nil && !os.IsNotExist(err) {
+		return "", errors.Wrapf(err, "error removing %s for container %s", destFile, c.ID())
+	}
+
 	f, err := os.Create(destFileName)
 	if err != nil {
 		return "", errors.Wrapf(err, "unable to create %s", destFileName)
 	}
-
 	defer f.Close()
-	_, err = f.WriteString(output)
-	if err != nil {
+	if err := f.Chown(c.RootUID(), c.RootGID()); err != nil {
```

Comment:
I preferred to not add much more local state if not required, also because IIRC there cannot be more than 5 mappings (hard coded in the kernel) and usually the mapping with root is the first one, so we need just one iteration in the for loop.  In any case, if you prefer I can change it

Reply:
Alright, if it's guaranteed to be a quick check there's no point in storing it.

Reply:
The 5 mappings max has been removed in upstream kernels, but it is not likely to be long.